src/EasyAbp.BookingService.Domain/EasyAbp/BookingService/TimeInAdvance.cs

```diff
@@ -49,12 +51,64 @@ public class TimeInAdvance : ValueObject
     /// Null values have the lowest priority.
     /// </summary>
     public TimeSpan? MinTimespanInAdvance { get; set; }
-    
+
     protected override IEnumerable<object> GetAtomicValues()
     {
         yield return MaxDaysInAdvance;
         yield return MaxTimespanInAdvance;
         yield return MinDaysInAdvance;
         yield return MinTimespanInAdvance;
     }
+
+    public bool CanBook(TimeSpan timeSpan)
+    {
+        if (MaxDaysInAdvance == -1 || MaxTimespanInAdvance == TimeSpan.Zero)
```

Comment:
`-1` is only an example in the remark, not a strict agreement.

```csharp
    /// <summary>
    /// The maximum number of days people can occupy assets in advance.
    /// Given we are occupying from "March 10, 2022, 10:00".
    /// Value <c>3</c> means can occupy AFTER "March 7, 2022, 0:00".
    /// Value <c>0</c> means can occupy AFTER "March 10, 2022, 0:00".
    /// Value <c>-1</c> means can not occupy.
    /// Only the value with an EARLIER time of this property and the <see cref="MaxTimespanInAdvance"/> property will take effect.
    /// </summary>
```

When its value is `-1`, we can only book "March 10, 2022, 10:00" after "March 11, 2022, 0:00".
This must be impossible, so we don't check `MaxDaysInAdvance == -1` or `MaxDaysInAdvance < 0` here.